src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets

```diff
@@ -0,0 +1,370 @@
+<!--
+***********************************************************************************************
+Microsoft.NET.ComposeCache.targets
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+Copyright (c) .NET Foundation. All rights reserved. 
+***********************************************************************************************
+-->
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <!--
+    ============================================================
+                                        ComposeCache
+ 
+    The main publish entry point.
+    ============================================================
+    -->
+  <PropertyGroup>
+    <MSBuildAllProjects>$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
+  </PropertyGroup>
+
+  <Target Name="ComposeCache"
+          DependsOnTargets="PrepareForComposeCache;
+                            Restore;
+                            ComputeAndCopyFilesToCacheDirectory"/>
+
+  <!--
+    ============================================================
+                                        PrepareForComposeCache
+
+    Prepare the prerequisites for ComposeCache.
+    ============================================================
+    -->
+  <Target Name="PrepareForComposeCache">
+
+    <PropertyGroup>
+      <PreserveCacheLayout Condition="'$(PreserveCacheLayout)' == ''">true</PreserveCacheLayout>
+      <PublishDir>$(ComposeDir)</PublishDir>
+      <PublishDir Condition="'$(ComposeDir)' == ''">$(OutputPath)</PublishDir>
+      <PublishDir Condition="'$(ComposeDir)' == '' and '$(RuntimeIdentifier)' != ''">$(OutputPath)$(RuntimeIdentifier)\</PublishDir>
+      
+      <FX_Version Condition="'$(FX_Version)' == ''">$(_ShortFrameworkVersion)</FX_Version>
+      <_CrossProjFileDir>$([System.IO.Path]::Combine($([System.IO.Path]::GetTempPath()),"Optimize_Crossgen_$(RuntimeIdentifier)_$(TargetFramework)_$(FX_Version)"))</_CrossProjFileDir>
```

Comment:
Is there any reason not to use a folder under the intermediate output path for this?